ReactWindows/ReactNative/Views/WebView/ReactWebViewManager.cs

```diff
@@ -0,0 +1,289 @@
+                        else return;
+                    }
+                    else request.Method = HttpMethod.Get;
+
+                    if (source.TryGetValue("headers", out headers) && headers.Type == JTokenType.Object)
+                    {
+                        IEnumerator<KeyValuePair<string, JToken>> enumerator = ((JObject)headers).GetEnumerator();
+                        enumerator.Reset();
+                        while (enumerator.MoveNext())
+                        {
+                            if (enumerator.Current.Value.Type == JTokenType.String)
+                            {
+                                request.Headers.Append(enumerator.Current.Key, enumerator.Current.Value.Value<string>());
+                            }
+                        }
+                    }
+
+                    if (source.TryGetValue("body", out body) && body.Type == JTokenType.String)
+                    {
+                        request.Content = new HttpStringContent(body.Value<string>());
+                    }
+
+                    view.NavigateWithHttpRequestMessage(request);
+                    return;
+                }
+            }
+            
+            view.Navigate(new Uri(BLANK_URL));
+        }
+
+        /// <summary>
+        /// Receive events/commands directly from JavaScript through the 
+        /// <see cref="UIManagerModule"/>.
+        /// </summary>
+        /// <param name="view">
+        /// The view instance that should receive the command.
+        /// </param>
+        /// <param name="commandId">Identifer for the command.</param>
+        /// <param name="args">Optional arguments for the command.</param>
+        public override void ReceiveCommand(Windows.UI.Xaml.Controls.WebView view, int commandId, JArray args)
+        {
+            switch (commandId)
+            {
+                case CommandGoBack:
+                    if (view.CanGoBack) view.GoBack();
+                    break;
+
+                case CommandGoForward:
+                    if (view.CanGoForward) view.GoForward();
+                    break;
+
+                case CommandReload:
+                    view.Refresh();
+                    break;
+
+                default:
+                    throw new InvalidOperationException(
+                        $"Unsupported command '{commandId}' received by '{typeof(ReactWebViewManager)}'.");
+            }
+        }
+
+        /// <summary>
+        /// Called when view is detached from view hierarchy and allows for 
+        /// additional cleanup by the <see cref="ReactWebViewManager"/>.
+        /// </summary>
+        /// <param name="reactContext">The react context.</param>
+        /// <param name="view">The view.</param>
+        public override void OnDropViewInstance(ThemedReactContext reactContext, Windows.UI.Xaml.Controls.WebView view)
+        {
+            view.NavigationCompleted -= OnNavigationCompleted;
```

Comment:
Sounds good, just curious.

---
In reply to: [60718668](https://github.com/ReactWindows/react-native/pull/372#discussion_r60718668) [](ancestors = 60718668)